version: '3'
services:
  mysql:
    image: mysql:5.7.26
    restart: always
    environment:
      MYSQL_ONETIME_PASSWORD: "true"
      MYSQL_ROOT_PASSWORD: "${MYSQL_ROOT_PASSWORD}"
      MYSQL_DATABASE: "${MYSQL_DATABASE}"
      MYSQL_USER: "${MYSQL_USER}"
      MYSQL_PASSWORD: "${MYSQL_PASSWORD}"
    volumes:
      - shinobi_mysql:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost", "-u$MYSQL_USER", "-p$MYSQL_PASSWORD"]
      timeout: 20s
      retries: 10
    command: --innodb-buffer-pool-size=1G --innodb-log-buffer-size=128M

  shinobi:
    depends_on:
      - mysql
    build: .
    restart: always
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      - ./config:/config
      - ./videos:/opt/shinobi/videos
      - /dev/shm/shinobiDockerTemp:/dev/shm/streams
    ports:
      - "${SHINOBI_PORT:-8080}:8080"
    environment:
      MYSQL_HOST: "mysql"
      MYSQL_DATABASE: "${MYSQL_DATABASE}"
      MYSQL_USER: "${MYSQL_USER}"
      MYSQL_PASSWORD: "${MYSQL_PASSWORD}"
      ADMIN_USER: "${ADMIN_USER}"
      ADMIN_PASSWORD: "${ADMIN_PASSWORD}"
      utcOffset: "${utcOffset}"

volumes:
  shinobi_mysql: {}
